src/Model/OidcExchangeTokens.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Drenso\OidcBundle\Model;
+
+use stdClass;
+
+class OidcExchangeTokens extends OidcTokens
```

Comment:
I'm not a particular fan of this construct, as it implies that the id token is available in the class. I propose a different solution, which is splitting the OidcTokens class into two. Something like this:

```php
class OidcTokens extends UnvalidatedOidcTokens
{
  public function __construct(stdClass $tokens) {
    // validates the existence of the id and access tokens and calls the parent constructor
  }

  // Overrides for the accessor methods to remove the nullable option
}

class UnvalidatedOidcTokens {
  // This would be the original OidcTokens class without the token isset validation
  // The id/access token properties should become nullable and protected, the accessor methods should return nullable strings
}
```

Reply:
Good idea. `UnvalidatedOidcTokens` is now the base class of `OidcTokens`. I had to do a minimal signature change: The constructor of both classes originally wanted an `\stdClass`, now it's `object`. That way I can also pass an `UnvalidatedOidcTokens` instance when converting it to a validated one.